modules/tracking/samples/lasot_benchmark.py

```diff
@@ -0,0 +1,249 @@
+    precision = np.sqrt((new_cx - gt_cx) ** 2 + (new_cy - gt_cy) ** 2)
+    return precision
+
+
+def get_norm_pr(new, gt, gt_bb_w, gt_bb_h):
+    '''
+    In calculations of precision and normalized precision are used thresholds
+    from original TrackingNet paper: for metric calculation TrackingNet using
+    lists of thresholds, but here we use only one numerical value of the
+    threshold for each metric
+    Lists, generated by 'calc_coords', are used as input for these methods.
+
+    Respectively:
+    new[0],gt[0] -> xmin, new[1],gt[1] -> xmax, new[2],gt[2] -> ymin
+    new[3],gt[3] -> ymax, new[4],gt[4] -> cx, new[5],gt[5] -> cy
+    '''
+    new_xmin, new_xmax, new_ymin, new_ymax, new_cx, new_cy = new
+    gt_xmin, gt_xmax, gt_ymin, gt_ymax, gt_cx, gt_cy = gt
+    normalized_precision = np.sqrt(((new_cx - gt_cx) / gt_bb_w) ** 2 + (
+        (new_cy - gt_cy) / gt_bb_h) ** 2)
+    return normalized_precision
+
+
+def init_tracker(tracker_name):
+    '''
+    Method used for initializing of trackers by creating it
+    via cv.TrackerX_create().
+    Input: string with tracker name.
+    Output: dictionary 'config'
+    Dictionary 'config' contains trackers names
+    as keys and tuple with call method and number of frames for
+    reinitialization as values
+    '''
+    config = {"Boosting": (cv.TrackerBoosting_create(), 500),
+              "MIL": (cv.TrackerMIL_create(), 1000),
+              "KCF": (cv.TrackerKCF_create(), 1000),
+              "MedianFlow": (cv.TrackerMedianFlow_create(), 1000),
+              "GOTURN": (cv.TrackerGOTURN_create(), 250),
+              "MOSSE": (cv.TrackerMOSSE_create(), 1000),
+              "CSRT": (cv.TrackerCSRT_create(), 1000)}
+    return config[tracker_name]
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run LaSOT-based benchmark for visual object trackers")
+    # As a default argument used name of
+    # original dataset folder
+    parser.add_argument("--path_to_dataset", type=str,
+                        default="LaSOTTesting", help="Full path to LaSOT")
+    parser.add_argument("--visualization", action='store_true',
+                        help="Showing process of tracking")
+    args = parser.parse_args()
+
+    # Creating list with names of videos via reading names from txt file
+    video_names = os.path.join(args.path_to_dataset, "testing_set.txt")
+    with open(video_names, 'rt') as f:
+        list_of_videos = f.read().rstrip('\n').split('\n')
+    trackers = [
+        'Boosting', 'MIL', 'KCF', 'MedianFlow', 'GOTURN', 'MOSSE', 'CSRT']
+
+    list_iou = []
+    list_pr = []
+    list_n_pr = []
+
+    # Loop for every tracker
+    for tracker_name in trackers:
+
+        print("Tracker name: ", tracker_name)
+
+        iou_video = np.zeros(21)
+        pr_video = np.zeros(21)
+        n_pr_video = np.zeros(21)
+        iou_thr = np.linspace(0, 1, 21)
+        pr_thr = np.linspace(0, 50, 21)
+        n_pr_thr = np.linspace(0, 0.5, 21)
```

Comment:
What is `21`?

Reply:
Number of thresholds, used for integration.

Reply:
Please add a named constant for it.

Reply:
OK.